server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/NodeStats.java

```diff
@@ -112,6 +116,7 @@ public NodeStats(StreamInput in) throws IOException {
         ingestStats = in.readOptionalWriteable(IngestStats::read);
         adaptiveSelectionStats = in.readOptionalWriteable(AdaptiveSelectionStats::new);
         indexingPressureStats = in.readOptionalWriteable(IndexingPressureStats::new);
+        repositoriesStats = in.readOptionalWriteable(RepositoriesStats::new);
```

Comment:
We should handle the mixed-node serialization here too. That's the reason why the bwc tests are failing.